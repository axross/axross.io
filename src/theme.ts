import Theme from './core/Theme';

const theme: Theme = {
  spacing: {
    small: 8,
    regular: 16,
    large: 24,
    huge: 40,
    enormous: 80,
  },
  typography: {
    headline1: {
      family: 'Rubik',
      size: 96,
      weight: 700,
    },
    headline2: {
      family: 'Rubik',
      size: 60,
      weight: 700,
    },
    headline3: {
      family: 'Rubik',
      size: 48,
      weight: 700,
    },
    headline4: {
      family: 'Rubik',
      size: 34,
      weight: 700,
    },
    headline5: {
      family: 'Rubik',
      size: 24,
      weight: 700,
    },
    headline6: {
      family: 'Rubik',
      size: 20,
      weight: 500,
    },
    body1: {
      family: 'Rubik',
      size: 16,
      weight: 400,
    },
    body2: {
      family: 'Rubik',
      size: 14,
      weight: 400,
    },
    caption: {
      family: 'Rubik',
      size: 12,
      weight: 400,
    },
  },
  // nanka
  // typography: {
  //   headline1: {
  //     family: 'Merriweather',
  //     size: 96,
  //     weight: 900,
  //     italic: true,
  //   },
  //   headline2: {
  //     family: 'Merriweather',
  //     size: 60,
  //     weight: 300,
  //   },
  //   headline3: {
  //     family: 'Merriweather',
  //     size: 48,
  //     weight: 900,
  //     italic: true,
  //   },
  //   headline4: {
  //     family: 'Merriweather',
  //     size: 34,
  //     weight: 400,
  //   },
  //   headline5: {
  //     family: 'Merriweather',
  //     size: 24,
  //     weight: 400,
  //   },
  //   headline6: {
  //     family: 'Merriweather',
  //     size: 20,
  //     weight: 900,
  //     italic: true,
  //   },
  //   body1: {
  //     family: 'Merriweather',
  //     size: 16,
  //     weight: 400,
  //   },
  //   body2: {
  //     family: 'Merriweather',
  //     size: 14,
  //     weight: 400,
  //   },
  //   caption: {
  //     family: 'Merriweather',
  //     size: 12,
  //     weight: 400,
  //   },
  // },
};

export default theme;
