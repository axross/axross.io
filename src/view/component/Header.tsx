import { ClassAttributes, createElement } from 'react';
import styled from '../../core/emotion';
import Text from './Text';

type Props = ClassAttributes<HTMLElement> & {
  name: string;
  title: string;
  className?: string;
};

const Header = ({ name, title, className }: Props) => (
  <Root className={className}>
    <Name type="headline1">{name}</Name>

    <Title type="headline3">{title}</Title>
  </Root>
);

const Root = styled('header')`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto auto;
  grid-template-areas: 'name' 'title';
  align-content: center;
  width: 100%;
  height: 100vh;
`;

const Name = styled(Text)`
  grid-area: name;
  margin-bottom: ${({ theme }) => theme.spacing.regular}px;
`;

const Title = styled(Text)`
  grid-area: title;
`;

export default Header;
