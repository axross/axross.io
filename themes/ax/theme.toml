# theme.toml template for a Hugo theme
# See https://github.com/spf13/hugoThemes#themetoml for an example

name = "Ax"
license = "MIT"
licenselink = "https://github.com/axross/axross.io/blob/master/LICENSE.md"
description = ""
homepage = "http://axross.io/"
tags = []
features = []
min_version = 0.19

[author]
  name = "axross"
  homepage = "http://axross.io"
