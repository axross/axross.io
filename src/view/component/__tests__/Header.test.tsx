import { ThemeProvider } from 'emotion-theming';
import { createElement } from 'react';
import { create } from 'react-test-renderer';
import theme from '../../../theme';
import Header from '../Header';

describe('<Header>', () => {
  test('it matches with the previous snapshot', () => {
    const renderer = create(
      <ThemeProvider theme={theme}>
        <Header className="amsterdam" />
      </ThemeProvider>,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
