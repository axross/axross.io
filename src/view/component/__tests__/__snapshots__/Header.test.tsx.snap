// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> it matches with the previous snapshot 1`] = `
.emotion-2 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto auto;
  grid-template-areas: 'name' 'title';
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  width: 100%;
  height: 100vh;
}

.emotion-0 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Rubik;
  -webkit-font-smoothing: antialiased;
  font-size: 96px;
  font-weight: 700;
  font-style: normal;
  grid-area: name;
  margin-bottom: 16px;
}

.emotion-1 {
  display: block;
  line-height: 1;
  color: #fff;
  font-family: Rubik;
  -webkit-font-smoothing: antialiased;
  font-size: 48px;
  font-weight: 700;
  font-style: normal;
  grid-area: title;
}

<header
  className="amsterdam emotion-2"
>
  <span
    className="emotion-0"
  >
    MITCH GRASSI
  </span>
  <span
    className="emotion-1"
  >
    tenor lead and backing vocals
  </span>
</header>
`;
